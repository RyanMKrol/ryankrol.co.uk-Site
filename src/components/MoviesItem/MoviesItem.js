import React, { Component } from 'react';

import styles from './MoviesItem.module.css';

/**
 * A component representing a books item
 */
class MoviesItem extends Component {
  /**
   * @class
   * @param {object} props arguments to class
   */
  constructor(props) {
    super();

    this.name = props.Title;
    this.thumbnail = props.Poster;
    this.runtime = props.Runtime;
    this.releaseDate = props.Released;

    if (
      typeof this.name === 'undefined'
      || typeof this.thumbnail === 'undefined'
      || typeof this.runtime === 'undefined'
      || typeof this.releaseDate === 'undefined'
    ) {
      throw new Error('Could not create MoviesItem');
    }
  }

  /**
   * Render method
   *
   * @returns {React.Component} JSX MoviesItem component
   */
  render() {
    return (
      <div className={styles.container}>
        <img className={styles.thumbnail} src={this.thumbnail} alt={this.name} />
      </div>
    );
  }
}

export default MoviesItem;
