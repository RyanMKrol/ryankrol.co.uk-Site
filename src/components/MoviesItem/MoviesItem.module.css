.thumbnail,
.container {
  max-height: 180px;
}
