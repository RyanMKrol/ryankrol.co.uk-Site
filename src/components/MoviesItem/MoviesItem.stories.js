import React from 'react';

import DEFAULT_MOCK from './__mocks__/MoviesItemMocks';

export default {
  title: 'Components/MoviesItem',
};

/**
 * default story
 *
 * @returns {React.Component} JSX for the MoviesItem
 */
export const normal = () => DEFAULT_MOCK;
