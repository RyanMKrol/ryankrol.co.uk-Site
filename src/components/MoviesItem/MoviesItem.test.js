import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';

import MoviesItem from './MoviesItem';

import DEFAULT_MOCK from './__mocks__/MoviesItemMocks';

it('renders MoviesItem without issue', () => {
  const listensItem = shallow(DEFAULT_MOCK);
  expect(toJson(listensItem)).toMatchSnapshot();
});

it('renders MoviesItem without throwing', () => {
  expect(() => mount(DEFAULT_MOCK)).not.toThrow();
});

it('MoviesItem throws without name', () => {
  expect(() => mount(<MoviesItem />)).toThrow();
});

it('MoviesItem throws without thumbnail', () => {
  expect(() => mount(<MoviesItem name={'About a Boy'} />)).toThrow();
});

it('MoviesItem throws without runtime', () => {
  expect(() => mount(<MoviesItem name={'About a Boy'} runtime={'101 min'} />)).toThrow();
});

it('MoviesItem throws without releaseDate', () => {
  expect(() => mount(
      <MoviesItem
        name={'About a Boy'}
        thumbnail={
          'https://m.media-amazon.com/images/M/MV5BMTQ2Mzg4MDAzNV5BMl5BanBnXkFtZTgwMjcxNTYxMTE@._V1_SX300.jpg'
        }
        runtime={'101 min'}
      />,
  )).toThrow();
});
