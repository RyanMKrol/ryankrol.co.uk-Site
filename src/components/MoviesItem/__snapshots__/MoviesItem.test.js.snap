// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders MoviesItem without issue 1`] = `
<div
  className="container"
>
  <img
    alt="About a Boy"
    className="thumbnail"
    src="https://m.media-amazon.com/images/M/MV5BMTQ2Mzg4MDAzNV5BMl5BanBnXkFtZTgwMjcxNTYxMTE@._V1_SX300.jpg"
  />
</div>
`;
